.NET/Microsoft.Recognizers.Text/Utilities/RegexCache.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Recognizers.Text
+{
+    public static class RegexCache
+    {
+        private static ConcurrentDictionary<(string pattern, RegexOptions options), Regex> _cache = new ConcurrentDictionary<(string pattern, RegexOptions options), Regex>();
+
+        public static bool Compiled { get; set; } = false;
```

Comment:
How would you set the Compiled flag? Only at code build time? Otherwise most will already be in memory and not compiled at init time.